File: frontend/src/api/appConfig.ts
import { http } from './client'

/**
 * UI 可编辑的运行时配置(LLM 中转/对象存储/语音/BGM 等密钥)。
 * RuntimeConfig 优先读 DB,空时回落 application-local.yml。
 */

export interface AppConfigItem {
  configKey: string
  description: string | null
  secret: boolean
  category: string
  sortOrder: number
  /** DB 里是否有值。true=用户已在 UI 填过(或 autoImport 自动迁移过)*/
  hasDbValue: boolean
  /** 显示值。secret=true 时已被后端 mask 成 abcd****wxyz 形式 */
  displayValue: string
  updatedAt: string | null
}

export async function listAppConfig(): Promise<AppConfigItem[]> {
  const { data } = await http.get<AppConfigItem[]>('/config')
  return data
}

/**
 * 批量保存。kv 是 { key: value } 对;value="" 表示清空走 yml 兜底。
 * secret 字段值若是 mask 占位(abcd****wxyz),前端不应传过来 — 只把"用户改过的"键值传 PUT。
 */
export async function updateAppConfig(kv: Record<string, string>): Promise<{ updated: number; keys: string[] }> {
  const { data } = await http.put<{ updated: number; keys: string[] }>('/config', kv)
  return data
}

File: backend/src/main/java/com/auteur/runtimeconfig/AppConfig.java
package com.auteur.runtimeconfig;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDateTime;

/**
 * UI 可编辑的运行时配置(KV 表)。
 *
 * 跟 application-local.yml 配合工作:RuntimeConfig 优先读本表,空时回落 yml 默认值。
 * 新用户的 yml 只需配 spring.datasource.*,其它全部在前端「系统设置」页面填。
 *
 * is_secret=true 的字段(api-key 等)REST 响应里会被遮掉,只保留是否已配置标志。
 */
@Entity
@Table(name = "app_config")
@Getter
@Setter
public class AppConfig {

    @Id
    @Column(name = "config_key", length = 128)
    private String configKey;

    /** 实际值。NULL/空 = 未配置,走 yml 兜底。 */
    @Column(name = "config_value", columnDefinition = "TEXT")
    @JsonIgnore  // 默认不直出原值;controller 自己决定 mask 还是直出
    private String configValue;

    @Column(name = "description", length = 512)
    private String description;

    @Column(name = "is_secret", nullable = false)
    private boolean secret;

    /** 分组,UI 用:llm / tos / voice / bgm / extension */
    @Column(name = "category", nullable = false, length = 64)
    private String category;

    @Column(name = "sort_order", nullable = false)
    private int sortOrder;

    @UpdateTimestamp
    @Column(name = "updated_at", nullable = false)
    private LocalDateTime updatedAt;
}
